src/elements/unified-share-modal-element/UnifiedShareModalElement.js

```diff
@@ -0,0 +1,172 @@
+    const { apiHost, itemID, itemType, language, token }: USMElementProps = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?USMElementSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorExists, setErrorExists] = useState<boolean>(false);
+
+    const api = new API({
+        apiHost,
+        clientName: CLIENT_NAME_USM_ELEMENT,
+        id: `${itemType}_usm`,
+        token,
+    });
+
+    const resetState = useCallback(() => {
+        setItem(null);
+        setSharedLink(null);
+        setCurrentUserID(null);
+    }, [setItem, setSharedLink, setCurrentUserID]);
+
+    const getError = useCallback(() => {
+        setErrorExists(true);
+    }, [setErrorExists]);
+
+    const getUserSuccess = useCallback(
+        userData => {
+            const { id, userEnterpriseData } = normalizeUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorExists(false);
+        },
+        [sharedLink],
+    );
+    const getItemSuccess = (itemData: USMElementItemAPIResponse) => {
+        const { item: itemFromAPI, sharedLink: sharedLinkFromAPI } = normalizeItemResponse(itemData);
+        setItem(itemFromAPI);
+        setSharedLink(sharedLinkFromAPI);
+        setErrorExists(false);
+    };
+
+    useEffect(() => {
+        resetState();
+    }, [resetState, token, itemID, itemType]);
+
+    useEffect(() => {
+        const getUserData = async () => {
+            await api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
+                fields: [FIELD_ENTERPRISE, FIELD_HOSTNAME],
+            });
+        };
+        if (item && sharedLink && !currentUserID) {
+            getUserData();
+        }
+    });
+
+    useEffect(() => {
+        const getItem = async () => {
+            if (itemType === TYPE_FILE) {
+                await api.getFileAPI().getFile(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_TYPE,
+                    ],
+                });
+            }
+
+            if (itemType === TYPE_FOLDER) {
+                await api.getFolderAPI().getFolderFields(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_TYPE,
```

Comment:
I know what this means, but the name is a bit misleading in this context. The field name is "type" but this could be misconstrued as "a type of field".